### ISSUE TITLE
feat(infra): Automatically docs deployment with CI.

### ISSUE THREAD

**Issue by yecol** [apache/incubator-graphar#569]
### Describe the enhancement requested

As I inspect the docs.yml and if I have not overlooked something, the action seems only check and build the website?

Maybe we can improve the CI to make the website automatically built and... Deployed!
The practice in graphscope website maybe work:

- in this repo, docs.yml checks if anything in /docs changes, if yes, either:
    - in repo incubator-graphar-website, keep a synchronized copy of `/docs`, push the change to that repo, and if any change happens, build and deploy with CI. 
    - or even built the website `/docs` in this repo and push to the branch `asf-site` of repo  incubator-graphar-website directly

using this action: https://github.com/ad-m/github-push-action

### Component(s)

Developer Tools

**Comment by acezen** [comment]
Does github support pushing change to another repo in action?

BTW, the [Github Actions Policy](https://infra.apache.org/github-actions-policy.html) may give some help about how to use third-party action.

**Comment by yecol** [comment]
I see. 
Although the Apache github actions policy allows to use third-party action, (seems it only requires to specify the hash?)
I get your concerns that this solution may have some security issue.

Alternatively, how about this semi-automatic solution?
- to set the deploy CI in `incubator-graphar-website` as scheduled nightly: if any changed in `/docs` then deploy.

**Comment by acezen** [comment]
> I see. Although the Apache github actions policy allows to use third-party action, (seems it only requires to specify the hash?) I get your concerns that this solution may have some security issue.
> 
> Alternatively, how about this semi-automatic solution?
> 
> * to set the deploy CI in `incubator-graphar-website` as scheduled nightly: if any changed in `/docs` then deploy.

yes, we do have a scheduled nightly [workflow](https://github.com/apache/incubator-graphar-website/blob/main/.github/workflows/devdoc.yml) in `incubator-graphar-website`, but the check change step has some problem and not work as expected :(
